src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -455,13 +456,34 @@ protected int AddMetadataReferenceAndTryConvertingToProjectReferenceIfPossible(s
                 }
             }
 
-            if (!File.Exists(filePath))
-            {
-                return hResultForMissingFile;
-            }
-
+            // regardless whether the file exists or not, we still record it. one of reason we do that is some cross language p2p references might be resolved
+            // after they are already reported as metadata references. since we use bin path as a way to discover them, if we don't previously record the reference ourselves, 
+            // cross p2p references won't be resolved as p2p references when we finally have all required information.
+            //
+            // it looks like 1. project system sometimes won't guarantee build dependency for intellisense build if it is cross language dependency
+            //               2. output path of referenced cross language project might be changed to right one once it is already added as metadata reference.
+            //
+            // but this has one concequence. even if a user adds a project in the solution as a metadata reference explicitly, that dll will be automatically 
+            // converted back to p2p reference. unfortunately there is no way to prevent this using information we have since, at this point, we don't know whether 
+            // it is a metadata reference added because we don't have enough information yet for p2p reference or user explicitly added it as a metadata reference.
             AddMetadataReferenceCore(this.MetadataReferenceProvider.CreateMetadataReference(this, filePath, properties));
```

Comment:
n/m.  looks like you doc things fully here.  BTW, please wrap at 100 columns for comments.